extensions/functions_datetime.yaml

```diff
@@ -48,6 +117,8 @@ scalar_functions:
             value: timestamp_tz
           - name: y
             value: interval_year
+          - name: timezone
+            value: string
```

Comment:
I don't see why a timezone is necessary for adding intervals to timestamp_tz (or timestamp). Coincidentally, I wrote about what I feel like the semantics should be yesterday here: https://github.com/substrait-io/substrait/pull/247#discussion_r968290105, but please correct me if those semantics are insufficient.

Reply:
I believe we need them if we are adding a timestamp_tz and a calendrical unit (month in this case I think) because result will depend on wether we cross DST jump or not.

Reply:
You're right. (jeez, I made a *lot* of comments about this without considering the non-physical nature of interval_year)